qiskit/user_config.py

```diff
@@ -138,6 +139,72 @@ def read_config_file(self):
                 self.settings['num_processes'] = num_processes
 
 
+def set_config(key, value, section=None, file_path=None):
+    """Adds or modifies a user configuration
+
+    It will add the configuration in either the default location
+    ~/.qiskit/settings.conf or the value of file argument.
+
+    Only valid user config can be set in 'default' section. Custom
+    user config can be added in any other sections.
+
```

Comment:
We should add a note here that any changes to the existing config file will not be reflected in the current session since the config file is parsed at import time.